### DISCUSSION TITLE
Rustdesk's licensing

### DISCUSSION THREAD

**Discussion by 0xthrowawaygithubuser** [rustdesk/rustdesk#1676] [General]
Hello to the Rustdesk community,
Rustdesk is currently licensed under [AGPL V3](https://github.com/rustdesk/rustdesk/blob/master/LICENCE). While GPL licenses are great, they make Rustdesk almost unusable in an enterprise production environment, because any modification to either the Rustdesk client or server will have to be GPLed and open-sourced, which is not viable for most enterprise applications and especially SaaS products that want to include a modified version of Rustdesk into their offering.

GPL licenses usually cause no issues when used as code libraries because usually no modifications are required, and even if there are required modifications, usually it's not a problem to open-source them because the libraries constitute a small fraction of the enterprise's IP.

However, in end-to-end products such as Ruskdesk, modification often is required, for example, to add logging and monitoring, set custom authentication, etc...
And the modified version could be the entire / a big part of the company's IP, so open-sourcing is unviable, especially when factoring in the security risks of open-sourcing a hosted service that is not well-reviewed by security researchers (such as an open-source modified version of Rustdesk).
And ultimately, exposing a version of Rustdesk that was modified for a specific application at a specific company will contribute very little to the open-source community.

I think Rustdesk is great software, and in order for it to be used by even more people for more applications, such as commercial use, I suggest it will be licensed under a more permissive license, such as MIT, that allows modified versions of Rustdesk to remain proprietary.

Thanks!

**Comment by test2a** [comment]
@0xthrowawaygithubuser i am not criticizing you or anything, just don't take it that way.

GPL code is often mistaken by people. you start with a wrong premise and come to a wrong conclusion.


here,
"While GPL licenses are great, they make Rustdesk almost unusable in an enterprise production environment, because any modification to either the Rustdesk client or server will have to be GPLed and open sourced, which is not viable for most enterprise applications and especially SaaS products that want to include a modified version of Rustdesk into their offering."

> you have to provide your modified source code to your customers, essentially, you CANNOT use rustdesk code to build proprietary software.

that is the just of using GPL over MIT code. MIT license tells you "do whatever you want, including turn this MIT code into proprietary code".  MIT code gives "freedom to developers" who can decide if the end users get benefit of open source, the 4 freedoms or not.
GPL on the other hand, ensures "freedom to end user" and that is a big deal for choosing this license.

>such as MIT, that allows modified versions of Rustdesk to remain proprietary.
this is exactly why the devs have chosen GPL over MIT, they are driven by the 4 freedoms and i for one support them. The basic argument of "free software" is to avoid proprietary code and rustdesk is doing its important part in this. 
I don't think the devs will change the license, they probably cannot even if they wanted to because there are many people contributing to the project and they have done so KNOWING this project is and will remain AGPL. 

Short of being a dev, this is the best i could do.

**Comment by 123survesh** [comment]
@0xthrowawaygithubuser if you are not going to sell your code or release it to the public as a new product, I believe you are free to do your custom updates to it and use it. I don't think you have to publish your code online. So for personal use, even in a big organization it should be fine. (please correct me if I am wrong)

Speaking as a libre enthusiast, I believe the people chose a GPL based license to avoid the purposes that you are mentioning. A libre software contains the time and effort of a lot of people and they get to choose what others can do with the product they created, just like how your organization has its agendas and goals with the software that you write.

GPL is a cool way of saying hey pay back the gratitude in code, not just money and this is a software for the world at all times and we don't like it when others try to keep the public from accessing our code.

And MIT vs GPL is a very age old debate, I would suggest reading up on why GPL was created they way it is (https://www.gnu.org/licenses/rms-why-gplv3.html is a good read)

**Comment by selfisekai** [comment]
@rustdesk I need a clarification for the license for this project - is the license SPDX specifier [`AGPL-3.0-or-later`](https://spdx.org/licenses/AGPL-3.0-or-later.html) (that is, if FSF releases a new version of AGPL, the new version may also be used by the users), or [`AGPL-3.0-only`](https://spdx.org/licenses/AGPL-3.0-only.html)?

**Reply by test2a** [reply]
usually its -or later one

**Comment by Darin755** [comment]
I find this discussion to be a bit offensive to me personally. What you saying is you want to seal the code from a community project and keep it for yourself.

Propeitary software is a method of unjust control. I know many will disagree but that doesn't mean that the communitys code should go towards a propeitary code base that could be malware

**Comment by JsonSong89** [comment]
This repo is about "**remote controll**" , not some base tools like sdk,web framework . 
I don't think MIT is suitable for this project.

**Comment by adamency** [comment]
@0xthrowawaygithubuser You have a lot of nerve lol. Stealing FOSS code to make easy money on its merits. If you want to make commercial use, use the budget you have to create an alternative solution. GPL has been precisely created to protect against the likes of you.

**Comment by My1** [comment]
you could just write your code in a way that doesnt put the confidential data into code, so you can e.g. put the IP addresses into a setting or whatever, so you dont need to publish those.
